base/schedulers/CDLL.jl

```diff
@@ -0,0 +1,260 @@
+
+module ConcurrentList #Concurrent Doubly Linked List
+
+mutable struct Node{T}
+    const value::Union{T, Nothing}
+    @atomic next::Union{Node{T}, Nothing}
```

Comment:
You can make this type stable by having `next` and `prev` point to the current node in case there is no next node.

Reply:
Oh cool. This will all need some fine comb optimizations for sure.